client/java/src/main/java/io/openlineage/client/transports/CompositeTransport.java

```diff
@@ -36,41 +40,83 @@ public List<Transport> getTransports() {
 
   @Override
   public void emit(@NonNull OpenLineage.RunEvent runEvent) {
-    for (Transport transport : transports) {
-      try {
-        transport.emit(runEvent);
-      } catch (Exception e) {
-        handleEmissionFailure(transport, e);
-      }
-    }
+    genericEmit(runEvent);
   }
 
   @Override
   public void emit(@NonNull OpenLineage.DatasetEvent datasetEvent) {
-    for (Transport transport : transports) {
-      try {
-        transport.emit(datasetEvent);
-      } catch (Exception e) {
-        handleEmissionFailure(transport, e);
-      }
-    }
+    genericEmit(datasetEvent);
   }
 
   @Override
   public void emit(@NonNull OpenLineage.JobEvent jobEvent) {
-    for (Transport transport : transports) {
+    genericEmit(jobEvent);
+  }
+
+  /**
+   * Emit events in parallel using a thread pool. Gets Object as param since different event types
+   * do not share a common base class nor interface.
+   *
+   * @param event
+   */
+  private void genericEmit(Object event) {
+    if (!config.getContinueOnFailure()) {
+      // Emit events sequentially
+      for (Transport transport : transports) {
+        emit(transport, event);
+      }
+    } else {
+      // Emit events in parallel
       try {
-        transport.emit(jobEvent);
-      } catch (Exception e) {
-        handleEmissionFailure(transport, e);
+        Executors.newCachedThreadPool()
```

Comment:
Do not create new thread pool on each emit. Create it on transport constructor, and maybe allow configuration of it.

Reply:
I've also included test to verify a pool is shutdown when needed.